internal/executor/instance/prebuilt.go

```diff
@@ -0,0 +1,177 @@
+package instance
+
+import (
+	"archive/tar"
+	"bufio"
+	"context"
+	"encoding/json"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/sirupsen/logrus"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type PrebuiltInstance struct {
+	Image      string
+	Dockerfile string
+	Arguments  map[string]string
+}
+
+func CreateTempArchive(dir string) (string, error) {
```

Comment:
We should look into reusing parts of `internal/targz/targz.go` from the agent here.

Reply:
`targz` name suggests that it wouldn't be compatible with Docker Engine API right away, since the latter expects a simple tar archive, so it would require some changes. Moreover, the implementation seems to be doing strange things like zeroing the modification timestamps, which is crucial for the correct operation of many things (`make`, etc).

Perhaps we should keep it as is since it's not that huge and does what it's designed to do, and not multiple things at once?

Reply:
I was thinking of refactoring `targz` in two methods. Something like https://github.com/cirruslabs/cirrus-ci-agent/pull/39

Reply:
What about the other issue?

Reply:
Seems that we've decided to keep this as is, at least for now.